### PR TITLE
Report error call-stack to New Relic 

### LINKED ISSUE #48: Report error call-stack to New Relic
Not happy with the way NewRelic displays error info - https://rpm.newrelic.com/accounts/1433338/applications/27979543/traced_errors/13af8f9e-e853-11e6-8137-f8bc12425498_1054_2142

If we want that we probably need to catch the stack trace (should be happening in error-handler.js anyway) and send it to new relic

https://docs.newrelic.com/docs/agents/nodejs-agent/supported-features/nodejs-agent-api#noticeError

See example how to do it here:
http://stackoverflow.com/questions/26964009/detecting-process-crashes-with-new-relic-in-nodejs

### PR DESCRIPTION
**Pull request by virtser**
closes #48

### CODE REVIEW

File: src/utils/serverRunner.js [hunk 1]
```diff
@@ -23,7 +24,14 @@ function handleProcessErrors(server) {
       Logger.close()
     ])
     .catch(() => {}) // ignore errors thrown here
-    .then(() => process.exit(-1));
+    .then(() => {
+      if (newrelic) {
+        newrelic.addCustomParameter('crash', 'true');
```
Comment (avnersorek) [inline src/utils/serverRunner.js]: Please put this code in the ``utils/newrelic`` util in a ``crashClose`` method (or something like that), which returns a promise and add that to the ``Promise.all`` above

Comment (virtser) [inline src/utils/serverRunner.js]: Done, please review.

File: src/utils/newrelic.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+function crashClose() {
+  if (!_newrelic) {
+    return Promise.resolve();
+  } else {
+    return new Promise(resolve => {
+      _newrelic.addCustomParameter('crash', 'true');
+      _newrelic.agent.harvest(() => {
+        process.exit(-1);
```
Comment (avnersorek) [inline src/utils/newrelic.js]: No need to ``process.exit`` here - just ``_newrelic.agent.harvest(resolve)``
``serverRunner`` will exit after everyone is ready

File: src/utils/newrelic.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      _newrelic = require('newrelic');
+    }
+  }  
+  return _newrelic;
+}
+
+function crashClose() {
+  if (!_newrelic) {
+    return Promise.resolve();
+  } else {
```
Comment (avnersorek) [inline src/utils/newrelic.js]: this is just a style issue but IMHO if you ``return`` in the ``if`` it's better not to use ``else`` - just reduces some nesting and less curly braces.

File: src/utils/newrelic.js (hunk 2 above)
Comment (virtser) [inline src/utils/newrelic.js]: Cool

File: src/utils/newrelic.js (hunk 3 above)
Comment (virtser) [inline src/utils/newrelic.js]: fixed

**Pull request merged**